datastore/longhorn.go

```diff
@@ -393,11 +393,13 @@ func (s *DataStore) ListSettings() (map[types.SettingName]*longhorn.Setting, err
 		// Cannot use cached object from lister
 		settingField := types.SettingName(itemRO.Name)
 		// Ignore the items that we don't recongize
-		if _, ok := types.SettingDefinitions[settingField]; ok {
+		if _, ok := types.GetSettingDefinition(settingField); ok {
 			itemMap[settingField] = itemRO.DeepCopy()
 		}
 	}
 	// fill up the missing entries
+	types.SettingDefinitionsLock.RLock()
```

Comment:
Could we just have a `GetSettingDefinitions` in setting.go? so we don't need to use a specific lock in setting.go, but rather focus on using the interfaces from types package only. (it will not 100% atomic, but it should be fine)

Reply:
Addressed. Thank you.